server/pkg/client/vault.go

```diff
@@ -84,3 +86,41 @@ func (v *Vault) GetCert(secretName, customerID string) (map[string]string, error
 
 	return certMap, nil
 }
+func (v *Vault) PostCredentials(secretName, username, password, astradbcreds string) error {
+	if v.client == nil {
+		return fmt.Errorf("Vault client is not initialized")
+	}
+	secretData := map[string]interface{}{
+		"username": username,
+		"password": password,
+	}
+
+	ctx := context.Background()
+
+	// Write the secret
+	_, err := v.client.KVv2(secretName).Put(ctx, astradbcreds, secretData)
```

Comment:
post credentials is generic function name, why is it astradb specific names passed